File: app/src/main/java/org/dhis2/usescases/main/MainComponent.java
package org.dhis2.usescases.main;


import org.dhis2.data.dagger.PerActivity;

import dagger.Subcomponent;

/**
 * Created by ppajuelo on 17/10/2017.
 */
@PerActivity
@Subcomponent(modules = MainModule.class)
public interface MainComponent {
    void inject(MainActivity mainActivity);
}

File: app/src/main/java/org/dhis2/usescases/main/MainComponent.kt
package org.dhis2.usescases.main

import dagger.Subcomponent
import org.dhis2.commons.di.dagger.PerActivity
import org.dhis2.usescases.troubleshooting.TroubleshootingComponent
import org.dhis2.usescases.troubleshooting.TroubleshootingModule
import org.dhis2.usescases.videoGuide.VideoGuideComponent
import org.dhis2.usescases.videoGuide.VideoGuideModule

@PerActivity
@Subcomponent(modules = [MainModule::class])
interface MainComponent {
    fun inject(mainActivity: MainActivity)
    fun plus(troubleShootingModule: TroubleshootingModule): TroubleshootingComponent
    fun plus(videoGuideModule: VideoGuideModule): VideoGuideComponent
}
